### ISSUE TITLE
[Bug]: Fluent v8 custom group headers DetailsList example table semantics

### ISSUE THREAD

**Issue by smhigley** [microsoft/fluentui#29769]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/customgroupheaders

### Bug Description

Example code only issue:

The Custom Group Headers example doesn't have the correct row/gridcell role info or ARIA attributes, so it has a broken a11y tree and broken SR table navigation in virtual cursor.

The fix would be to add the needed roles & ARIA attrs to the example's custom group header render fn.

### Requested priority

Low

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by jenshaw86** [comment]
@smhigley It appears that this issue isn't limited to custom group headers, but to default group headers too. SR table navigation works as expected until a group is collapsed/expanded.

**Comment by smhigley** [comment]
Dropping some context here on @jenshaw86's comment: after investigating, we found the issue on the non-custom grouped DetailsList is a Narrator-only issue caused by Edge's UIA mappings + `aria-rowindex`. A bug has been logged on Edge, and documented in our [internal wiki](https://dev.azure.com/microsoftdesign/fluent-ui/_wiki/wikis/fluent-ui.wiki/332/Fluent-v8-known-issues)
